server/controllers/finance/reports/shared.js

```diff
@@ -0,0 +1,85 @@
+const _ = require('lodash');
+const PeriodService = require('../../../lib/period');
+
+const columns = [
+  { field : 'billingDateFrom', displayName : 'FORM.LABELS.DATE', comparitor : '>', isDate : true },
+  { field : 'billingDateTo', displayName : 'FORM.LABELS.DATE', comparitor : '<', isDate : true },
+  { field : 'cashbox_id', displayName : 'FORM.LABELS.CASHBOX' },
+  { field : 'cash_uuid', displayName : 'FORM.INFO.PAYMENT' },
+  { field : 'currency_id', displayName : 'FORM.LABELS.CURRENCY' },
+  { field : 'dateFrom', displayName : 'FORM.LABELS.DATE_FROM', comparitor : '>', isDate : true },
+  { field : 'dateTo', displayName : 'FORM.LABELS.DATE_TO', comparitor : '<', isDate : true },
+  { field : 'debtor_uuid', displayName : 'FORM.LABELS.CLIENT' },
+  { field : 'debtor_group_uuid', displayName : 'FORM.LABELS.DEBTOR_GROUP' },
+
+  { field : 'is_caution', displayName : 'FORM.LABELS.CAUTION' },
+  { field : 'supplier_uuid', displayName : 'FORM.LABELS.SUPPLIER' },
+  { field : 'status_id', displayName : 'PURCHASES.ORDER' },
+  { field : 'patientReference', displayName : 'FORM.LABELS.REFERENCE_PATIENT' },
+  { field : 'user_id', displayName : 'FORM.LABELS.USER' },
+
+  { field : 'invoiceReference', displayName : 'FORM.LABELS.INVOICE' },
+  { field : 'invoice_uuid', displayName : 'FORM.LABELS.INVOICE' },
+  { field : 'service_id', displayName : 'FORM.LABELS.SERVICE' },
+  { field : 'inventory_uuid', displayName : 'FORM.LABELS.INVENTORY' },
+
+  { field : 'flux_id', displayName: 'STOCK.FLUX' },
+  { field : 'status', displayName: 'STOCK.STATUS.LABEL' },
+  { field : 'hospital_no', displayName : 'FORM.LABELS.HOSPITAL_NO' },
+  { field : 'patient_group_uuid', displayName : 'PATIENT_GROUP.PATIENT_GROUP' },
+
+  { field : 'entry_date_from', displayName: 'STOCK.ENTRY_DATE', comparitor: '>', isDate : true },
+  { field : 'entry_date_to', displayName: 'STOCK.ENTRY_DATE', comparitor: '<', isDate : true },
+  { field : 'expiration_date_from', displayName: 'STOCK.EXPIRATION_DATE', comparitor: '>', isDate : true },
+  { field : 'expiration_date_to', displayName: 'STOCK.EXPIRATION_DATE', comparitor: '<', isDate : true },
+  { field : 'is_exit', displayName : 'STOCK.OUTPUT' },
+  { field : 'label', displayName : 'STOCK.LOT' },
```

Comment:
Now all filter field are in the same place, some of them lead to confusion, we should rename them appropriately even in an other pull request.

Reply:
I agree - most of the registry keys are unique, since many of them share the same filters.  However, the stock registries have many keys that overlap with other registries.  We could address this in another PR, since it will require changing both client and server.